src/condor_tools/htcondor_cli/annex.py

```diff
@@ -6,6 +6,129 @@
+    """
+    Summarize the usage of an annex.
+    """
+
+    options = {
+        "annex_name": {
+            "args": ("--annex-name",),
+            "help": "annex name",
+        },
+    }
+
+
+    # Unlike `shutdown`, status should probably not require a name.
+    def __init__(self, logger, **options):
+        the_annex_name = options["annex_name"];
+
+        schedd = htcondor.Schedd()
+        if the_annex_name is None:
+            annex_jobs = schedd.query(f'hpc_annex_name =!= undefined')
+        else:
+            annex_jobs = schedd.query(f'hpc_annex_name == "{the_annex_name}"')
+
+        # Each annex request is represented by its own job, but we want
+        # to present aggregate information for each annex name.
+
+        status = { job["hpc_annex_name"]: {} for job in annex_jobs }
+        for job in annex_jobs:
+            annex_name = job["hpc_annex_name"]
+            request_id = job["hpc_annex_request_id"]
+
+            status[annex_name][request_id] = "requested"
+            if job.get("hpc_annex_PID") is not None:
+                status[annex_name][request_id] = "submitting (1)"
+            if job.get("hpc_annex_PILOT_DIR") is not None:
+                status[annex_name][request_id] = "submitting (2)"
+            if job.get("hpc_annex_JOB_ID") is not None:
+                status[annex_name][request_id] = "submitted"
+            if job["JobStatus"] == htcondor.JobStatus.IDLE:
+                status[annex_name][request_id] = "granted"
+
+        annex_collector = htcondor.param.get("ANNEX_COLLECTOR", "htcondor-cm-hpcannex.osgdev.chtc.io")
+        collector = htcondor.Collector(annex_collector)
+        if the_annex_name is None:
+            annex_slots = collector.query(constraint=f'AnnexName =!= undefined', ad_type=htcondor.AdTypes.Startd)
+        else:
+            annex_slots = collector.query(constraint=f'AnnexName == "{the_annex_name}"', ad_type=htcondor.AdTypes.Startd)
+
+        for slot in annex_slots:
+            annex_name = slot["AnnexName"]
+            request_id = slot["hpc_annex_request_id"]
+            status[annex_name][request_id] = {}
+
+        for slot in annex_slots:
+            annex_name = slot["AnnexName"]
+            request_id = slot["hpc_annex_request_id"]
+
+            # Ignore dynamic slot, since we just care about aggregates.
+            # We could write a similar check for static slots, but we
+            # don't ever start any annexes with static slots.  (Bad
+            # admin, no cookie!)
+            if slot.get("PartitionableSlot", False):
+                increment(status, annex_name, request_id, "TotalCPUs", slot["TotalSlotCPUs"])
+                increment(status, annex_name, request_id, "BusyCPUs", len(slot["ChildCPUs"]))
+
+        # This could be substantially optimized with a projection.  It
+        # also assumes that the user can only see their own jobs.
+        running_jobs = {}
+        target_jobs = schedd.query(f'TargetAnnexName =!= undefined')
```

Comment:
I don't think this assumption works with the python bindings... unless I'm mistaken, `Schedd.query(constraint = X)` is just like `condor_q -l -const $X` back in the old days before you had to specify `-all`, regardless of the `CONDOR_Q_ONLY_MY_JOBS` setting. Maybe add a `--user` option that defaults to `getpass.getuser()` and do something like this?

```suggestion
        # This could be substantially optimized with a projection.  It
        # also assumes that the user can only see their own jobs.
        running_jobs = {}
        target_jobs = schedd.query(f'TargetAnnexName =!= undefined && Owner == "{options.user}"')
```

Reply:
After a quick test, it looks like you're right that the bindings are `-all` by default.  However, there is a convenient flag to do what we want:

schedd.query(opts=htcondor.QueryOpts.DefaultMyJobsOnly)`

Reply:
Perfect!